### PR TITLE
Add molecule tests, Jenkinsfile

### PR DESCRIPTION
**Pull request by nre-ableton**

### CODE REVIEW

**Review by mst-ableton — APPROVED** [review]
nice, lgtm!

File: requirements-dev.in [hunk 1]
```diff
@@ -0,0 +1,7 @@
+ansible
+ansible-lint
+molecule
+molecule-docker
+pytest-socket
+pytest-testinfra
+python-jenkins
```
Comment (ala-ableton) [inline requirements-dev.in:7]: `ansible-lint` and `molecule` are directly used in `Jenkinsfile`, but why do we list the other packages as first-level dependencies?

File: molecule/default/tests/test_default.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import os
+
+import testinfra.utils.ansible_runner
+
+from jenkins import Jenkins
+
+
+testinfra_hosts = testinfra.utils.ansible_runner.AnsibleRunner(
+    os.environ['MOLECULE_INVENTORY_FILE']).get_hosts('all')
```
Comment (ala-ableton) [inline molecule/default/tests/test_default.py:9]:
```suggestion
testinfra_hosts = testinfra.utils.ansible_runner.AnsibleRunner(
    os.environ['MOLECULE_INVENTORY_FILE']
).get_hosts('all')
```

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        // https://github.com/ansible-community/ansible-lint/issues/236
+        if (stdout) {
+          error 'ansible-lint exited with warnings, check the output of the previous step'
+        }
+      },
+      groovylint: { groovylint.check('./Jenkinsfile') },
+      molecule: { data.venv.run('molecule --debug test') },
+    )
+  },
+  deployWhen: { runTheBuilds.isPushTo(['main']) && env.PRODUCTION == 'true' },
```
Comment (ala-ableton) [inline Jenkinsfile:33]:
```suggestion
  deployWhen: { devToolsProject.shouldDeploy(defaultBranch: 'main') },
```

**Review by ala-ableton — COMMENTED** [review]
@nre-ableton sorry for being late on this one.

File: requirements-dev.in (hunk 1 above)
Comment (nre-ableton) [reply requirements-dev.in:7]: Molecule requires a compatible version of Ansible, but doesn't depend on it directly.

As for packages like `pytest-*` and `molecule-docker`, these are things that molecule can use, but again, doesn't directly depend on these packages.

File: molecule/default/tests/test_default.py (hunk 2 above)
Comment (nre-ableton) [reply molecule/default/tests/test_default.py:9]: Let's instead add black...

**Comment by nre-ableton** [comment]
@ala-ableton See https://github.com/Ableton/ansible-role-amtool/pull/4

**Pull request merged**